File: development/NET/samples/client/AECondition/CS/MyAeSession.cs
using System;
using Softing.OPCToolbox.Client;
using Softing.OPCToolbox;

namespace AECondition
{
	public class MyAeSession : AeSession
	{

		#region Constructor
		//-----------------

		public MyAeSession(string url) : base(url)
		{
			EnableAeConditionsBySourceCompleted += new EnableAeConditionsBySourceEventHandler(HandleEnableAeConditionsBySourceCompleted);

		}

		//--
		#endregion

		#region Private Members
		//---------------------

		#region Private Attributes
		//------------------------

		private OpcForm m_opcForm = null;

		//--
		#endregion

		//--
		#endregion

		#region Public Methods
		//---------------------

		public void SetForm(OpcForm form)
		{
			m_opcForm = form;
		}

		//--
		#endregion

		#region Public Properties
		//-----------------------


		//--
		#endregion

		#region Handles
		//---------------------

		private void HandleEnableAeConditionsBySourceCompleted(
			AeSession anAeSession,
			uint executionContext,
			bool enable,
			string path,
			int result)
		{
			if (ResultCode.SUCCEEDED(result))
			{
				Console.WriteLine("Enabling conditions succeeded!");
			}
		}

		//--
		#endregion

	}   //	end class MyAeSession

}   //	end namespace

File: development/NET/samples/client/AECondition/VB/MyAeSession.vb
Imports System
Imports Softing.OPCToolbox.Client
Imports Softing.OPCToolbox

Namespace AECondition
	Public Class MyAeSession
		Inherits AeSession

#Region "Constructor"
		'-----------------

		Public Sub New(ByVal url As String)
			MyBase.New(url)
			AddHandler EnableAeConditionsBySourceCompleted, AddressOf HandleEnableAeConditionsBySourceCompleted

		End Sub

		'--
#End Region

#Region "Private Members"
		'---------------------

#Region "Private Attributes"
		'------------------------

		Private m_opcForm As OpcForm = Nothing

		'--
#End Region

		'--
#End Region

#Region "Public Methods"
		'---------------------

		Public Sub SetForm(ByVal form As OpcForm)
			m_opcForm = form
		End Sub

		'--
#End Region

#Region "Public Properties"
		'-----------------------


		'--
#End Region

#Region "Handles"
		'---------------------

		Private Sub HandleEnableAeConditionsBySourceCompleted(ByVal anAeSession As AeSession, ByVal executionContext As UInt32, ByVal enable As Boolean, ByVal path As String, ByVal result As Integer)
			If ResultCode.SUCCEEDED(result) Then
				Console.WriteLine("Enabling conditions succeeded!")
			End If
		End Sub

		'--
#End Region

	End Class

End Namespace

